lib/internal/Magento/Framework/Reflection/TypeProcessor.php

```diff
@@ -708,4 +695,56 @@ private function getNormalizedType($type)
         }
         return $type;
     }
+
+    /**
+     * Parses `return` annotation from reflection method.
+     *
+     * @param MethodReflection $methodReflection
+     * @return ReturnTag
+     * @throws \InvalidArgumentException if doc block is empty or `@return` annotation doesn't exist
+     */
+    private function getMethodReturnAnnotation(MethodReflection $methodReflection)
+    {
+        $methodName = $methodReflection->getName();
+        $returnAnnotations = $this->getReturnFromDocBlock($methodReflection);
+        if (empty($returnAnnotations)) {
+            // method can inherit doc block from implemented interface, like for interceptors
+            $implemented = $methodReflection->getDeclaringClass()->getInterfaces();
```

Comment:
Sir, you're missing docs from extended class here. $methodReflection->getDeclaringClass()->getParentClass() should be included in this array

Reply:
Hi @VoidWalker, please report an issue using a template consisting of full steps to reproduce.